clogMonitor/src/components/LogEventsFilters.js

```diff
@@ -82,6 +83,46 @@ const LogEventsFilters = ({ dataSetHandler }) => {
     // Datetime states (Dates stored are as local time strings, not UTC time)
     const [startTime, setStartTime] = React.useState(getDefaultDateTimeString(0));
     const [endTime, setEndTime] = React.useState(getDefaultDateTimeString(1));
+    
+    /* TEST NEEDED ********************** begin */
+
+    //DST confliction BEFORE or AFTER selection
+    const [startTimeDST, setStartTimeDST] = React.useState([]);
+    const [endTimeDST, setEndTimeDST] = React.useState([]);
+    const getStartDstDatetimeHandler = (setter) => {
+        function f(event) {
+            let val = event.target.value; //BEFORE or AFTER
+            setter(val);
+            let tm = startTimeDST.state; //time: '2021-11-07T01:30:00'
+            if(hasDSTconflict(tm)) {
+                if (val==='AFTER') { //
+                    let date = new Date(tm);
+                    let mdate = moment(date).add(1,'hours').format("YYYY-MM-DDTHH:mm:ss");
+                    setStartTimeDST(mdate);          
+                }
+            }
+        }
+        return f
+        //return (event) => setter(event.target.value);
+    }
+    const getEndDstDatetimeHandler = (setter) => {
```

Comment:
You should be able to use the same getHandler function for both the start and end dsts, the only difference is the setter, which is parameterized.